### ISSUE TITLE
😔 Notification pop-up appears while the App is open in Android

### ISSUE THREAD

**Issue by strawberry-code** [zo0r/react-native-push-notification#1411]
# Bug

On Android, when the App is open and a notification is received, I see that the notification pop-up appears, so, in the case the user taps the popup, the notification will be handled twice.

On iOS, when the App is open and a notification is received, I see no pop-ups and this is the correct behaviour I want.

Why on Android appears a notification pop-up when the App is open?

## Environment info

Library version: 3.1.9

```bash
System:
    OS: macOS 10.15.4
    CPU: (8) x64 Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz
    Memory: 2.36 GB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.15.0 - ~/.nvm/versions/node/v12.15.0/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v12.15.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.4, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK:
      API Levels: 26, 28, 29
      Build Tools: 28.0.3, 29.0.2
      System Images: android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom_64, android-28 | Google Play Intel x86 Atom_64, android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom_64, android-R | Google APIs Intel x86 Atom_64, android-R | Google Play Intel x86 Atom
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.6010548
    Xcode: 11.4.1/11E503a - /usr/bin/xcodebuild
  npmPackages:
    react: ^16.9.0 => 16.13.0 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    react-native-create-library: 3.1.2
```

**Comment by Dallas62** [comment]
Hi @strawberry-code 
What kind of notification ? remote, local, scheduled ?
Also, can't you check if the notification is already handled (workaround) ?

**Comment by strawberry-code** [comment]
Remote notification.

I tired to handle it, but the appearing of the pop up while App is open is causing problems of UX.

**Comment by Dallas62** [comment]
Ho, I didn't see you are on `3.1.9`, can you update please ?

**Comment by MaganAnkur** [comment]
strawberry-code I am getting the same behavior in iOS could you please let me know what can I do to stop it in iOS.

**Comment by strawberry-code** [comment]
@Dallas62 Ok, we will try to update it as soon as possible. We did a quick try locally to update library but linking the new library is giving us some problems on Android, idk why but we are investigating. I'll let you know as soon as possible if updating the library is the working solution.

@MaganAnkur Hi, unfortunately we have never encountered this problem on iOS. I'm not sure, but using the **popInitialNotification** flag set to **false** could be your solution?

**Comment by s-z-s** [comment]
@strawberry-code I am on 3.3.0 and this doesn't happen in my app. Updating will work for you.

**Comment by strawberry-code** [comment]
@Dallas62 great! Updating the library solved the problem of showing the notification pop-up while App is open... but: I see no more the pop-up when App is closed or in background. 

🤯

Practically, the problem has been inverted.

I hope posting here my configure could be helpful to find if there is something wrong:

```
const configure = () => {
    console.log('push notification configured');
    PushNotificationIOS.addEventListener('registrationError', (e) => {
        alert(JSON.stringify(e));
    });

    PushNotification.configure({

        onRegister: function (token) {
            //process token
            PushNotification.setApplicationIconBadgeNumber(0);
            console.log('🎉 Token! ' + JSON.stringify(token));
            db.setToken(token.token).catch(
                //console.log('device push token has been saved on the database'),
            );
        },

        onNotification: async notification => {
            PushNotification.setApplicationIconBadgeNumber(0);
            console.log('🎈 Notifica arrivata! (AppState: ' + AppState.currentState + ')\nNotification object: ' + JSON.stringify(notification));
           
            // process the notification
            ...

            // required on iOS only
            if (Platform.OS === 'ios') {
                notification.finish(PushNotificationIOS.FetchResult.NoData);
            }

        },
        senderID: '11223344',

        permissions: {
            alert: true,
            badge: true,
            sound: true,
        },

        popInitialNotification: true,
        requestPermissions: true,
    });
};
```

Why on Android I am not seeing the incoming notifications when App is in background/closed? In addition, I see no more notifications in Android notification center.

**Comment by Dallas62** [comment]
Sorry, but what is your version now?

**Comment by strawberry-code** [comment]
I updated to 3.3.1.

**Comment by Dallas62** [comment]
Did you clean the cache ? This should be good on this version

**Comment by strawberry-code** [comment]
I ran `watchman watch-del-all && react-native start --reset-cache`, rebuilt and reinstalled the App, but not working. 🙁

**Comment by Dallas62** [comment]
Can you provide the AndroidManifest?
Maybe he is not up to date

**Comment by strawberry-code** [comment]
Sorry for the late reply, here is my **AndroidManifest.xml**:

```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
          package="com.acme">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
    <uses-permission android:name="android.permission.USE_BIOMETRIC" />
    <uses-permission android:name="android.permission.USE_FINGERPRINT" />
    <uses-permission android:name="android.permission.INTENT_FILTER_VERIFICATION_AGENT" />

    <application
            android:name=".MainApplication"
            android:label="@string/app_name"
            android:icon="@mipmap/ic_launcher"
            android:roundIcon="@mipmap/ic_launcher_round"
            android:allowBackup="false"
            android:networkSecurityConfig="@xml/network_security_config"
            android:usesCleartextTraffic="true"
            android:theme="@style/AppTheme">
            <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
            </intent-filter>
        </receiver>
        <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService" />
        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
            android:exported="false">
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>
        <activity
                android:name=".MainActivity"
                android:launchMode="singleTask"
                android:label="@string/app_name"
                android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
                android:screenOrientation="portrait"
                android:windowSoftInputMode="adjustResize">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:scheme="acme" />
            </intent-filter>
            <intent-filter android:autoVerify="true">
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:host="acme.com" android:pathPrefix="/ext/deep" android:scheme="https" />
            </intent-filter>
            <intent-filter android:autoVerify="true">
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:host="acme.com" android:pathPrefix="/ext/deep" android:scheme="https" />
            </intent-filter>
        </activity>
        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
                    android:value="fcm_default_channel"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
                    android:value="ACME channel"/>
        <!-- Change the resource name to your App's accent color - or any other color you want -->
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
                    android:resource="@android:color/white"/>

        <!-- </Else> -->
    </application>

</manifest>
```

**Comment by Dallas62** [comment]
If you reinstalled the notification, this should not be a problem.
Did you uninstall, install or replace an existing application?

Also you can try this solution:

https://github.com/zo0r/react-native-push-notification/issues/1395#issuecomment-623787394

Setting the default channel, this will force high importance and vibrate.

**Comment by strawberry-code** [comment]
I tried to use the proposed solution in the the comment you cited, but I am not sure how to do it, now my AndroidManifest.xml has changed in this way:

```
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
                    android:value="fcm_default_channel"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
                    android:value="ACME channel"/>
        <meta-data
                android:name="com.google.firebase.messaging.default_notification_channel_id"
                android:value="rn-push-notification-channel-id-4-300" />
        <!-- Change the resource name to your App's accent color - or any other color you want -->
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
                    android:resource="@android:color/white"/>
```

But I see no Remote notifications popups, however they comes "in background" and they are correctly processed by the App.

**Comment by Dallas62** [comment]
Did you uninstall, install or replace an existing application?
Do you see any notification in the notification center (in the silent part) ?

**Comment by strawberry-code** [comment]
> Did you uninstall, install or replace an existing application?
Nope, I am using a development device on which I am using/installing/reinstalling/updating only the App I am developing. No touching/uninstalling/installing/replacing other Apps.

> Do you see any notification in the notification center (in the silent part) ?

This is what I get if I am using 3.1.9:

![IMAGE 2020-05-06 11:32:33](https://user-images.githubusercontent.com/18684103/81162458-80a9e500-8f8d-11ea-9429-1130e448d39b.jpg)

And this is what I get using 3.3.1:

![IMAGE 2020-05-06 11:32:31](https://user-images.githubusercontent.com/18684103/81162484-8a334d00-8f8d-11ea-9f4f-5fb953da9887.jpg)

But in both cases the notifications are correctly handled by the App (I can see they are coming thorough Metro logs).

P.S. Sorry for the screenshots, they are both taken using 3.1.9 (I downgraded, but kept the 3.3.1 in a separated branch) and in the second one I simply removed manually the notifications, because I have no time to switch to the 3.3.1 branch. I did it to make the idea.
